gcloud/resource_manager/client.py

```diff
@@ -0,0 +1,298 @@
+to use is ``get_project()``, which will return ``None`` if the project
+doesn't exist::
+
+    >>> from gcloud import resource_manager
+    >>> client = resource_manager.Client()
+    >>> project = client.get_project('purple-spaceship-123')
+    >>> project.number is None
+    False
+    >>> project = client.get_project('doesnt-exist')
+    >>> project is None
+    True
+"""
+
+
+from gcloud.credentials import get_credentials
+from gcloud.credentials import get_for_service_account_json
+from gcloud.credentials import get_for_service_account_p12
+from gcloud.exceptions import NotFound
+from gcloud.resource_manager.connection import Connection
+from gcloud.resource_manager.iterator import ProjectIterator
+from gcloud.resource_manager.project import Project
+
+
+class Client(object):
+    """Client to bundle configuration needed for API requests.
+
+    See
+    https://cloud.google.com/resource-manager/reference/rest/
+    for more information on this API.
+
+    Automatically get credentials::
+
+        >>> from gcloud.resource_manager import Client
+        >>> client = Client()
+
+    .. note::
+
+        Chances are you don't want to use the arguments in the constructor
+        but instead want one of the factory methods (like
+        :func:`gcloud.resource_manager.client.Client.from_service_account_json`
+        or similar).
+
+        Even more likely is that you want to use the Cloud SDK to get
+        credentials for these API calls (that is, run ``gcloud auth login``).
+
+    :type credentials: :class:`oauth2client.client.OAuth2Credentials` or
+                       :class:`NoneType`
+
+    :param credentials: The OAuth2 Credentials to use for the connection
+                        owned by this client. If not passed (and if no ``http``
+                        object is passed), falls back to the default inferred
+                        from the environment.
+
+    :type http: :class:`httplib2.Http` or class that defines ``request()``.
+    :param http: An optional HTTP object to make requests. If not passed, an
+                 ``http`` object is created that is bound to the
+                 ``credentials`` for the current object.
+    """
+    def __init__(self, credentials=None, http=None):
+        self._connection = None
+        self._credentials = credentials
+        self._http = http
+
+    @property
+    def connection(self):
+        """Property that creates a Connection object on-demand."""
+        if self._connection is None:
+            self._connection = self._create_connection(
+                credentials=self._credentials, http=self._http)
+        return self._connection
+
+    @staticmethod
+    def _create_connection(credentials=None, http=None):
+        """Create and return a new connection object."""
+        return Connection(credentials=credentials or get_credentials(),
+                          http=http)
+
+    @classmethod
+    def from_service_account_json(cls, json_credentials_path):
```

Comment:
Sure. Would be easier to do it as a pre-requisite than an afterthought though (makes this commit much smaller and makes the refactor commit involve 2 locations -- old and new -- instead of 3 -- old here, old there and new).